### PR TITLE
Update `controller-gen` to `v0.9.0`

### PR DESCRIPTION
**Pull request by chrismuellner**
# Description
CRD was generated using an outdated version of `controller-gen`.

Removed `+kubebuilder:validation:MaxProperties=1` annotation for `OneAgentSpec` which as incompatible.

## How can this be tested?
Upgrade `controller-gen` and check if the generated CRD is the same:
1. `go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.0`
2. `make manifests`

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -220,7 +220,7 @@ ifeq (, $(shell which controller-gen))
 	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
 	cd $$CONTROLLER_GEN_TMP_DIR ;\
 	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.0 ;\
```
Comment (mjgrzybek) [inline Makefile:223]:
```suggestion
	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.0 ;\
```

**Review by mjgrzybek — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (mjgrzybek) [reply Makefile]: btw, I don't think Makefile should install tools. I'd leave it just checking if tool is available. 
Now it only installs it, but doesn't check if already existing one is in correct version.
Update is required to be done manually anyway.

**Comment by chrismuellner** [comment]
Those checks are generated when the project is generated with `operator-sdk`.
May be different now than when this Makefile was first generated.

**Comment by 0sewa0** [comment]
> I don't think Makefile should install tools.
I agree, but I think the main purpose of it is for the CI

**Comment by mjgrzybek** [comment]
Ok, so now let's just bump version and we can discuss it when refactoring Makefile.

**Review by mjgrzybek — APPROVED** [review]

**Pull request merged**